### PR TITLE
postgres fix for circleci, add lint/lintfix scripts

### LINKED ISSUE #18: fix running tests in ci
Tests currently aren't running in ci.

One way we might fix this is add a `docker-compose.test.yml` that includes postgres as one of the services and then use that docker-compose file in ci.

This might be the easiest approach, but open to other options.

@kamicut

### PR DESCRIPTION
**Pull request by sethvincent**
This pr:
- fixes the circleci build by making sure the postgres db name is correct
- lints with devseed-standard via lint/lintfix npm scripts
- makes a number of linting fixes

There were a handful of interesting errors caught by the linter. I'll add comments for those and we may need to make a couple small revisions.

closes #18

### CODE REVIEW

File: app/manage/authz.js [hunk 1]
```diff
@@ -59,20 +59,19 @@ async function acceptToken (token, res, next) {
...
 /**
- * Protected routes first check the session for the user,
- * Get the associated accessToken from the database, and check for 
+ * Routes with `authenticate` middleware first check the session for the user,
+ * Get the associated accessToken from the database, and check for
  * the accessToken validity with hydra. If there isn't a session,
  * it checks for an Authorization header with a valid access token
  */
-async function protected (req, res, next) {
+async function authenticate (req, res, next) {
```
Comment (sethvincent) [inline app/manage/authz.js]: Apparently `protected` is a "protected" keyword so we're not supposed to use it.

File: app/oauth/consent.js [hunk 2]
```diff
@@ -12,9 +12,9 @@ async function idTokenExtraParams (sub) {
   const conn = await db()
   const [user] = await conn('users').where('id', sub)
   const { profile } = user
-  const displayName = profile.displayName || consent.subject
```
Comment (sethvincent) [inline app/oauth/consent.js]: `consent` is undefined here so presumably this never worked and it's fine to remove it?

File: app/oauth/login.js [hunk 3]
```diff
@@ -6,10 +6,10 @@ const hydra = require('../lib/hydra')
 const url = require('url')
 
 function getLogin (app) {
-  return async function login(req, res, next) {
+  return async function login (req, res, next) {
     const query = url.parse(req.url, true).query
     const challenge = query.login_challenge
-    if (!challenge) return next(e)
+    if (!challenge) return next()
```
Comment (sethvincent) [inline app/oauth/login.js]: `e` was undefined. Should we be defining an error here and passing that to `next`?

File: app/oauth/consent.js (hunk 2 above)
Comment (sethvincent) [reply app/oauth/consent.js]: the comment on the diff makes it look almost like the whole line is removed but its just ` || consent.subject` that is removed.

File: app/oauth/login.js (hunk 3 above)
Comment (kamicut) [reply app/oauth/login.js]: Not sure, but maybe we should mark it as a TODO?

File: app/oauth/consent.js (hunk 2 above)
Comment (kamicut) [reply app/oauth/consent.js]: It should be the argument to the function, `sub`

File: app/oauth/consent.js [hunk 4]
```diff
@@ -12,9 +12,9 @@ async function idTokenExtraParams (sub) {
...
-  const picture = path(['_xml2json', 'user', 'img', '@', 'href'], profile)
-    || `https://www.gravatar.com/avatar/${sub}?d=identicon`
+  const displayName = profile.displayName
```
Comment (sethvincent) [inline app/oauth/consent.js]:
```suggestion
  const displayName = profile.displayName || sub
```

File: app/oauth/login.js (hunk 3 above)
Comment (sethvincent) [reply app/oauth/login.js]: we'll make an issue for this

**Pull request merged**